docs(topojson-export): add topojson response documentation

Chwiggy opened the pull request:
### Pull Request Checklist
<!--- Please make sure you have completed the following items BEFORE submitting a pull request (put an x in each box
when you have checked you have done them): -->
- [ ] 1. I have [**rebased**][rebase] the latest version of the main branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes # .

### Information about the changes
- Reason for change: Updated documentation


[config]: https://GIScience.github.io/openrouteservice/run-instance/configuration/
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/main/CONTRIBUTE.md#pull-request-guidelines

Chwiggy committed:
docs(topojson-export): add topojson response documentation

takb committed:
feat: edge weights in JSON mode using less decimal places

koebi commented:
What about making the `json` response example smaller (i.e. 4 nodes and 2 edges, just to see what's happening) and adding a `topojson` response example for the same region, so people directly see the difference?

Commit:
feat: improve error message of topojson export

When requesting a topojson export, the accept-header must be set
to 'json' not 'topojson' because topojson is not yet an official
content type. Instead of just telling that topojson is invalid,
the error message now hints to using json instead.

koebi commented on ors-api/src/main/java/org/heigit/ors/api/responses/export/json/JsonEdge.java:
```diff
@@ -2,9 +2,11 @@
 import io.swagger.v3.oas.annotations.media.Schema;
 import org.heigit.ors.common.Pair;
 
+import java.text.DecimalFormat;
```
Why is this import needed here?

takb replied:
got forgotten somehow, removed

takb committed:
feat: edge weights in TopoJSON mode using less decimal places

takb committed:
docs(topojson-export): more detailed TopoJSON mode description

takb commented:
> What about making the `json` response example smaller (i.e. 4 nodes and 2 edges, just to see what's happening) and adding a `topojson` response example for the same region, so people directly see the difference?

I tried to find a smaller selection where the difference is visible. 

While doing so I discovered a potential bug in the TopoJSON creation, will need to look a bit further tomorrow...

koebi commented on docs/public/topojson_qgis.png:
This adds an image which is not used?

takb replied:
it is

koebi commented on docs/.vitepress/theme/index.ts:
```diff
@@ -1,11 +1,18 @@
-import { Theme } from 'vitepress';
+import { Theme, useData, useRoute } from 'vitepress';
 import VersionSwitcher from "../components/VersionSwitcher.vue";
+import codeblocksFold from 'vitepress-plugin-codeblocks-fold'; // import method
+import 'vitepress-plugin-codeblocks-fold/style/index.css'; // import style
```
what is this plugin used for?
Does that influence other codeblocks, and if so, do we want that?

koebi replied:
Update:
It influences *all* codeblocks, see new comment.

koebi commented on docs/run-instance/configuration/engine/profiles/build.md:
```diff
@@ -132,7 +132,7 @@ Properties beneath `ors.engine.profiles.*.ext_storages`:
 | TrailDifficulty        | object | Returns the trail difficulty in the route response, compatible with walking and cycling profiles                                                 |                                                   |
 | Wheelchair             | object | Wheelchair-specific attributes compatible only with that profile                                                                                 | [Wheelchair](#wheelchair)                         |
-| OsmId                  | object | Returns the OsmId of the way, compatible only with wheelchair profile                                                                            |                                                   |
+| OsmId                  | object | Returns the OsmId of the way, compatible with any profile type                                                                                   |                                                   |
```
Does this fix an issue with the docs, or is this only true when exporting the graph?

takb replied:
what it said in the docs was plain wrong

Commit:
feat: improve error message of TopoJSON export

The message now clarifies that the TopoJSON output format of the
export endpoint requires, both, accept header and content type
to be specified as application/json.

Commit:
refactor: improve export request preprocessing

Moving more towards the principle "parse, don't validate"
enables simplifying the conversion of the API-level
request into a core-level request and prepares factoring out
common behavior of multiple endpoints.

koebi commented on docs/.vitepress/theme/index.ts:
```diff
@@ -1,11 +1,18 @@
+    setup() {
+        const { frontmatter } = useData();
+        const route = useRoute();
+        codeblocksFold({ route, frontmatter }, true, 250);
```
```suggestion
        codeblocksFold({ route, frontmatter }, false, 250);
```

I'd rather we use `false` here, and fold codeblocks manually that we want folded, via the frontmatter:
https://github.com/T-miracle/vitepress-plugin-codeblocks-fold?tab=readme-ov-file#extended-use

takb replied:
ok

takb committed:
feat: improve TopoJSON fallback without OSM IDs present

takb committed:
docs(topojson-export): better examples, some structure fixes

takb committed:
docs(topojson-export): fixes
